core/domain/opportunity_services.py

```diff
@@ -189,8 +189,65 @@ def add_new_exploration_opportunities(story_id, exp_ids):
     """
     story = story_fetchers.get_story_by_id(story_id)
     topic = topic_fetchers.get_topic_by_id(story.corresponding_topic_id)
-    explorations = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+    _create_exploration_opportunities(story, topic, exp_ids)
+
+
+def create_exploration_opportunities_for_story(story_id, topic_id):
+    """Creates exploration opportunities corresponding to the supplied story ID
+    iff the topic linked to the story is published.
+
+    Args:
+        story_id: str. The ID of the story domain object.
+        topic_id: str. The ID of the topic domain object corresponding to the
+            supplied story.
+
+    Raises:
+        Exception. A topic with the given ID doesn't exist.
+        Exception. The topic rights could not be found.
+    """
+    story = story_fetchers.get_story_by_id(story_id)
+    topic = topic_fetchers.get_topic_by_id(topic_id)
+    topic_rights = topic_fetchers.get_topic_rights(topic.id, strict=False)
+    if topic_rights is None:
+        raise Exception(
+            'Could not find topic rights for topic ID: %s.' % topic.id)
+    if topic_rights.topic_is_published:
+        exp_ids_in_story = story.story_contents.get_all_linked_exp_ids()
+        _create_exploration_opportunities(story, topic, exp_ids_in_story)
+
+
+def create_exploration_opportunities_for_topic(topic_id):
+    """Creates exploration opportunities corresponding to the supplied topic ID
+    each of the corresponding topic's published stories.
+
+    Args:
+        topic_id: str. The ID of the topic domain object.
+    """
+    topic = topic_fetchers.get_topic_by_id(topic_id)
```

Comment:
DO we need to validate whether the topic is published?

Reply:
We can as an additional defensive check (this method should only be called from publish_topic). I'll update the docstring to reflect that the input topic should be published for now.